src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -646,6 +646,52 @@ type internal TypeCheckInfo
         let items = items |> RemoveExplicitlySuppressed g
         items, nenv.DisplayEnv, m
 
+    /// Find union cases and compatible active patterns when the item under the cursor position is an identifier with the type of a DU.
+    let GetUnionCasesAndActivePatternsEnvironmentLookupResolutionsAtPosition cursorPos =
+        let rec doesActivePatternTakeTypeAsInput g ty paramType =
+            match paramType with
+            | TType_var (typar, _) ->
+                match typar.Solution with
+                | Some paramType ->
+                    doesActivePatternTakeTypeAsInput g ty paramType
+                | _ -> false
+            | TType_fun (domainType, rangeType, _) ->
+                if typeEquiv g domainType ty then
+                    true
+                else
+                    doesActivePatternTakeTypeAsInput g ty rangeType
+            | _ -> false
+
+        let resolutions = GetCapturedNameResolutions cursorPos ResolveOverloads.Yes
+
+        if resolutions.Count = 1 then
+            let res = resolutions[0]
+
+            match res.Item with
+            | Item.Value vref ->
+                let ty = vref.Type
+                let nenv = res.NameResolutionEnv
+                let m = res.Range
+
+                match tryTcrefOfAppTy nenv.DisplayEnv.g ty with
+                | ValueSome tcRef when tcRef.IsUnionTycon ->
+                    let isUnionInScopeAsUnqualified = nenv.eTyconsByAccessNames.ContainsKey tcRef.DisplayName
+                    let cases = ResolveUnionCasesOfType ncenv m res.AccessorDomain ty tcRef
+                    let activePatterns =
+                        nenv.ePatItems
```

Comment:
Just use a computed list expression `[ for x in nenv.ePatItems do .... ]` to replace these pipelines